themes/bootstrap3/js/facets.js

```diff
@@ -149,6 +149,106 @@ function collapseTopFacets() {
 
 /* --- Side Facets --- */
 VuFind.register('sideFacets', function SideFacets() {
+  function addJSTreeListener(treeNode) {
+    treeNode.bind('ready.jstree', function onReadyJstree() {
+      var tree = $(this);
+      // if hierarchical facet contains 2 or less top level items, it is opened by default
+      if (tree.find('ul > li').length <= 2) {
+        tree.find('ul > li.jstree-node.jstree-closed > i.jstree-ocl').each(function openNode() {
+          tree.jstree('open_node', this, null, false);
+        });
+      }
+      // open facet if it has children and it is selected
+      $(tree.find('.jstree-node.active.jstree-closed')).each(function openNode() {
+        tree.jstree('open_node', this, null, false);
+      });
+    });
+  }
+
+  function loadAjaxSideFacets() {
+    var $container = $('.side-facets-container-ajax');
+    if ($container.length === 0) {
+      return;
+    }
+
+    var facetList = [];
+    var $facets = $container.find('div.collapse.in[data-facet], .checkbox-filter[data-facet]');
+    $facets.each(function addFacet() {
+      if (!$(this).data('loaded')) {
+        facetList.push($(this).data('facet'));
+      }
+    });
+    if (facetList.length === 0) {
+      return;
+    }
+    var urlParts = window.location.href.split('?');
+    var query = urlParts.length > 1 ? urlParts[1] : '';
+    var request = {
+      method: 'getSideFacets',
+      searchClassId: $container.data('searchClassId'),
+      query: query,
+      enabledFacets: facetList
+    };
+    $container.find('.facet-load-indicator').removeClass('hidden');
+    $.getJSON(VuFind.path + '/AJAX/JSON?' + query, request)
+      .done(function onGetSideFacetsDone(response) {
+        $.each(response.data.facets, function initFacet(facet, facetData) {
+          var containerSelector = typeof facetData.checkboxCount !== 'undefined'
+            ? '.checkbox-filter' : ':not(.checkbox-filter)';
+          var $facetContainer = $container.find(containerSelector + '[data-facet="' + facet + '"]');
+          $facetContainer.data('loaded', 'true');
+          if (typeof facetData.checkboxCount !== 'undefined') {
+            if (facetData.checkboxCount !== null) {
+              $facetContainer.find('.avail-count').text(
+                facetData.checkboxCount.toString().replace(/\B(?=(\d{3})+\b)/g, VuFind.translate('number_thousands_separator'))
+              );
+            }
+          } else if (typeof facetData.html !== 'undefined') {
+            $facetContainer.html(facetData.html);
+          } else {
+            // TODO: this block copied from facets.js, refactor
```

Comment:
@EreMaijala, is this TODO comment really still in need of doing? This is facets.js, and I don't see a duplicate of this code.

Reply:
It wasn't a direct duplicate anymore, but I've now cleaned up what was duplicated.